### PR TITLE
Handle changes in docker compat mode

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/podman/issues/12830

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: test/apiv2/20-containers.at
Comment (jwhonce) [inline test/apiv2/20-containers.at:421]:
```suggestion
cparam="repo=newrepo&changes=EXPOSE=9090"
t POST "commit?container=${cid:0:12}&$cparam" 201 \
  .Id~[0-9a-f]\\{64\\}
```

**Review by jwhonce — CHANGES_REQUESTED** [review]

File: test/apiv2/20-containers.at
Comment (rhatdan) [reply test/apiv2/20-containers.at]: Thanks.

**Comment by mheon** [comment]
LGTM once tests go green

**Comment by mheon** [comment]
Though, I would like to verify that this doesn't change the Libpod API as well, we have a habit of reusing the compat handlers a lot.

File: pkg/api/handlers/compat/images.go
Comment (edsantiago) [inline pkg/api/handlers/compat/images.go:141]: Test failure suggests that this is not doing the right thing when the `Changes` field is empty. I'm juggling too much this morning to delve deeper into it.

Comment (edsantiago) [reply pkg/api/handlers/compat/images.go:141]: My first approach would be to look into `nil` vs `[]` (empty array) vs `""` (empty string), see how `strings.Fields()` differs from `strings.Split()`. HTH.

Comment (edsantiago) [reply pkg/api/handlers/compat/images.go:141]: try `s/Println/Printf("%q\n")/`:
```
[]
[""]
```

Comment (rhatdan) [reply pkg/api/handlers/compat/images.go:141]: Yikes.

**Comment by edsantiago** [comment]
`70-short-names.at` now passes on my laptop. LGTM, assuming that `!= ""` is functionally equivalent to `!= nil` in this use case.

**Comment by edsantiago** [comment]
Tests green. @jwhonce please see if this addresses your review comments.

**Review by jwhonce — APPROVED** [review]

**Comment by leahneukirchen** [comment]
Sorry, I only noticed this PR after it got merged already: but why do you split on `,`?

If you send this to the official Docker daemon, you get:
`{"message":"cmd=foo,expose=9090 is not a valid change command"}`

If you send `changes=CMD foo,EXPOSE 9090`, it will set CMD to `foo,EXPOSE 9090`.

The correct way is to send `changes=CMD foo\nEXPOSE9090`, i.e.
`changes=CMD%20foo%0aEXPOSE%209090`

**Comment by rhatdan** [comment]
I used the docker command against this and it worked.  But you know this better then me. So I can change it to break on \n.

**Comment by leahneukirchen** [comment]
`docker commit --change 'WORKDIR /' --change 'EXPOSE 9999'` sends `changes=WORKDIR%20&changes=EXPOSE%209090`, what did you run?

**Comment by leahneukirchen** [comment]
Note that https://docs.docker.com/engine/api/v1.41/#operation/ImageCommit says "instructions to apply while committing", so having them newline-separated seems natural.

**Comment by leahneukirchen** [comment]
https://github.com/moby/moby/blob/6e05fa7076670a55d94c57578a3b0097f3ca4b53/builder/dockerfile/builder.go#L329

**Pull request merged**
